### PR TITLE
Remove workarounds needed for Scala 2.12.17 / scala-xml 2.1.0 / sbt

### PR DESCRIPTION
**Pull request by mkurz**
Introduced in #11435

Can be dropped as soon as sbt releases a version with it's Scala version upgraded (= 2.12.17+, where scala-xml 2.x gets pulled in), see https://github.com/sbt/sbt/pull/7021, probably this will be sbt 1.8.0-M1

### CODE REVIEW

**Comment by mkurz** [comment]
sbt 1.8 breaks our scripted tests, see #11527

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
       cmd: |
         if [ "$CACHE_HIT_COURSIER" = "false" ]; then
           sbt +update                                             # Runs with adoptium:8 (default)
-          # sbt --sbt-version 1.7.2 +update                       # If we run scripted tests with multiple sbt versions, we could init that sbt installs here
+          # sbt --sbt-version 1.8.0-RC1 +update                   # If we run scripted tests with multiple sbt versions, we could init that sbt installs here
```
Comment (georgeorfanidi) [inline .github/workflows/build-test.yml:48]:
```suggestion
          # sbt --sbt-version 1.8.0 +update                   # If we run scripted tests with multiple sbt versions, we could init that sbt installs here
```

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -145,7 +145,7 @@ jobs:
       scala: 2.13.x
       add-dimensions: >-
         {
-          "sbt": [ "1.7.2" ],
+          "sbt": [ "1.8.0-RC1" ],
```
Comment (georgeorfanidi) [inline .github/workflows/build-test.yml:148]:
```suggestion
          "sbt": [ "1.8.0" ],
```

File: documentation/manual/hacking/Translations.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ translation-project
 `build.properties` should contain the sbt version, ie:
 
 ```
-sbt.version=1.7.2
+sbt.version=1.8.0-RC1
```
Comment (georgeorfanidi) [inline documentation/manual/hacking/Translations.md:43]:
```suggestion
sbt.version=1.8.0
```

File: documentation/manual/releases/release29/migration29/Migration29.md [hunk 4]
```diff
@@ -20,13 +20,13 @@ Where the "x" in `2.9.x` is the minor version of Play you want to use, for insta
 
 ### sbt upgrade
 
-Play 2.9 only supports sbt 1.7. To update, change your `project/build.properties` so that it reads:
+Play 2.9 only supports sbt 1.8. To update, change your `project/build.properties` so that it reads:
 
 ```properties
-sbt.version=1.7.2
+sbt.version=1.8.0-RC1
```
Comment (georgeorfanidi) [inline documentation/manual/releases/release29/migration29/Migration29.md:26]:
```suggestion
sbt.version=1.8.0
```

File: documentation/manual/releases/release29/migration29/Migration29.md [hunk 5]
```diff
@@ -20,13 +20,13 @@ Where the "x" in `2.9.x` is the minor version of Play you want to use, for insta
...
 ```
 
-At the time of this writing `1.7.2` is the latest version in the sbt 1.x family, you may be able to use newer versions too. Check the release notes for both Play's minor version [releases](https://github.com/playframework/playframework/releases) and sbt's [releases](https://github.com/sbt/sbt/releases) for details.
+At the time of this writing `1.8.0-RC1` is the latest version in the sbt 1.x family, you may be able to use newer versions too. Check the release notes for both Play's minor version [releases](https://github.com/playframework/playframework/releases) and sbt's [releases](https://github.com/sbt/sbt/releases) for details.
```
Comment (georgeorfanidi) [inline documentation/manual/releases/release29/migration29/Migration29.md:29]:
```suggestion
At the time of this writing `1.8.0` is the latest version in the sbt 1.x family, you may be able to use newer versions too. Check the release notes for both Play's minor version [releases](https://github.com/playframework/playframework/releases) and sbt's [releases](https://github.com/sbt/sbt/releases) for details.
```

File: documentation/project/build.properties [hunk 6]
```diff
@@ -1,4 +1,4 @@
 # Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 
 # sync with project/build.properties
-sbt.version=1.7.2
+sbt.version=1.8.0-RC1
```
Comment (georgeorfanidi) [inline documentation/project/build.properties:4]:
```suggestion
sbt.version=1.8.0
```

File: project/build.properties [hunk 7]
```diff
@@ -1,4 +1,4 @@
 # Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 
 # sync with documentation/project/build.properties
-sbt.version=1.7.2
+sbt.version=1.8.0-RC1
```
Comment (georgeorfanidi) [inline project/build.properties:4]:
```suggestion
sbt.version=1.8.0
```

**Review by felipebonezi — COMMENTED** [review]
Is there any migration needed for `SBT 1.8`? If so, we could provide a link to help some who needs to migrate from 1.7 to 1.8.

**Comment by mkurz** [comment]
This pull request is finished, however depends on https://github.com/sbt/sbt/pull/7147. Locally everything works for me.

Last commit also implements `bgRun`:
```sh
$ sbt
[info] welcome to sbt 1.8.3-SNAPSHOT (Eclipse Adoptium Java 11.0.17)
...
[play-scala-seed] $ bgRun
Starting server

--- (Running the application, auto-reloading is enabled) ---

INFO  p.c.s.AkkaHttpServer - Listening for HTTP on /127.0.0.1:9000
[success] Total time: 1 s, completed Feb 9, 2023, 12:25:09 PM
[play-scala-seed] $ 

# App is running in the background now

[play-scala-seed] $ 

# Making first request to localhost:9000, following log appears:

INFO  p.a.h.HttpErrorHandlerExceptions - Registering exception handler: guice-provision-exception-handler
2023-02-09 12:25:14 INFO  play.api.http.EnabledFilters  Enabled Filters (see <https://www.playframework.com/documentation/latest/Filters>):

    play.filters.csrf.CSRFFilter
    play.filters.headers.SecurityHeadersFilter
    play.filters.hosts.AllowedHostsFilter
2023-02-09 12:25:14 INFO  play.api.Play  Application started (Dev) (no global state)
[play-scala-seed] $ 

# Let's see the running processes:

[play-scala-seed] $ ps
[info] JobHandle(1, bgRun, ProjectRef(uri("file:/home/mkurz/work/play-scala-seed/"), "root") / Compile / bgRun)
[success] Total time: 0 s, completed Feb 9, 2023, 12:25:21 PM
[play-scala-seed] $ 

# Again but with bgList:

[play-scala-seed] $ show bgList
[info] * JobHandle(1, bgRun, ProjectRef(uri("file:/home/mkurz/work/play-scala-seed/"), "root") / Compile / bgRun)
[success] Total time: 0 s, completed Feb 9, 2023, 12:25:28 PM
[play-scala-seed] $ 

# Now let's change a file and see if re-compiling works:

[play-scala-seed] $ 
[info] compiling 1 Scala source to /home/mkurz/work/play-scala-seed/target/scala-2.13/classes ...
                    
--- (RELOAD) ---
                    
2023-02-09 12:25:48 INFO  p.a.http.HttpErrorHandlerExceptions  Registering exception handler: guice-provision-exception-handler
2023-02-09 12:25:48 INFO  play.api.http.EnabledFilters  Enabled Filters (see <https://www.playframework.com/documentation/latest/Filters>):

    play.filters.csrf.CSRFFilter
    play.filters.headers.SecurityHeadersFilter
    play.filters.hosts.AllowedHostsFilter
2023-02-09 12:25:48 INFO  play.api.Play  Application started (Dev) (no global state)
[play-scala-seed] $ 

# Worked :) Again check the processes:

[play-scala-seed] $ ps
[info] JobHandle(1, bgRun, ProjectRef(uri("file:/home/mkurz/work/play-scala-seed/"), "root") / Compile / bgRun)
[success] Total time: 0 s, completed Feb 9, 2023, 12:25:52 PM

# Now let's shutdown process 1

[play-scala-seed] $ bgStop 1
Stopping server
[success] Total time: 0 s, completed Feb 9, 2023, 12:25:54 PM
2023-02-09 12:25:54 INFO  p.a.i.l.c.CoordinatedShutdownSupport  Starting synchronous coordinated shutdown with ServerStoppedReason reason and 35000 milliseconds timeout
2023-02-09 12:25:54 INFO  play.core.server.AkkaHttpServer  Stopping Akka HTTP server...
2023-02-09 12:25:54 INFO  play.core.server.AkkaHttpServer  Terminating server binding for /127.0.0.1:9000
2023-02-09 12:25:54 INFO  play.core.server.AkkaHttpServer  Running provided shutdown stop hooks

# Start app again:

[play-scala-seed] $  bgRun
Starting server

--- (Running the application, auto-reloading is enabled) ---

INFO  p.c.s.AkkaHttpServer - Listening for HTTP on /127.0.0.1:9000
[success] Total time: 1 s, completed Feb 9, 2023, 12:29:20 PM

# Let's try to start a second app:

[play-scala-seed] $ bgRun
Not starting server since one is already started
[success] Total time: 0 s, completed Feb 9, 2023, 12:29:23 PM

# Only one app allowed. Let's check if only one process is running (the second bgRun needs to be ignored)

[play-scala-seed] $ ps
[info] JobHandle(2, bgRun, ProjectRef(uri("file:/home/mkurz/work/play-scala-seed/"), "root") / Compile / bgRun)
[success] Total time: 0 s, completed Feb 9, 2023, 12:29:23 PM
[play-scala-seed] $ 

# Worked!
```

**Comment by mkurz** [comment]
Instead of opening up methods and keys in sbt (https://github.com/sbt/sbt/pull/7147), I moved `PlayRun` to the `sbt´ package. Originally I wanted to move the object to  `sbt.play.run`, but because of Scala's relative import feature there were too many clashes, so `sbt.PlayRun` will be fine.
I am not 100% happy, but ok with that now.

**Pull request merged**